### ISSUE TITLE
Design & Build Status Page

### ISSUE THREAD

**Issue by ststimac** [sonarwhal/sonarwhal.com#307]
Need a status page similar to https://www.traviscistatus.com/ to show how the scanner has been doing in the last few days. It will be updated every few minutes (maybe 15?). I think it should be publicly available. Some of the metrics:

- Graphs with the number of scans over time
- Graphs with the number of successful and failed scans over time (maybe it can be merged with the previous one)
- Graphs with the status of the queues over time and the number of workers working
- Average time to get a job started (since the moment it gets to the queue to the moment it starts scanning). Maybe current average and also graph with value over time.
- Average time to get a job finished (since the moment it starts to the moment we get all the results). Maybe current average and also graph with value over time.

@molant do we have a full list of metrics, or are we just going with the above from your previous email?

**Comment by ststimac** [comment]
This is the next item on my list. The 5 mentioned metrics above, is that all we have at the moment or do we have other metrics to show. 

For item #3 "Graphs with the status of the queues over time", what exactly is the data around that? What are the different queue statuses?

**Comment by molant** [comment]
> is that all we have at the moment or do we have other metrics to show.

We are storing everything so if you can come up with some other useful metric feel free to add it.

> "Graphs with the status of the queues over time", what exactly is the data around that? What are the different queue statuses?

We have a queue to start a scan (pending jobs), and another one to update the database. @sarvaje did I miss any?
The most relevant one is the one of pending jobs, no need to show the other one.

**Comment by sarvaje** [comment]
> We have a queue to start a scan (pending jobs), and another one to update the database. @sarvaje did I miss any?

No, those are the queue we use in the scanner. And as you say, the important is the pending jobs one.

**Comment by ststimac** [comment]
Sounds good. If I come up with any other ideas for metrics I'll let you know.

**Comment by ststimac** [comment]
**For the number of workers working metric:** 

1) do the same amount of workers work on each scan? if not then
2) is there a max amount of workers working on one scan? 

**Let me know if these make sense for break-out metrics:** 

Daily average # of workers working across all scans
Highest # of workers working at one time all scans

(Only applies if item 1 above isn't true) Average # of workers working on one scan
(Only applies if item 1 above isn't true) Highest # of workers working on one scan
